doc/glossary.rst

```diff
@@ -0,0 +1,1216 @@
+        model can be finalized by calling ``partial_fit()`` i.e. without
+        passing a further mini-batch.
+
+        Generally, estimator parameters should not be modified between calls
+        to ``partial_fit``, although ``partial_fit`` should validate them
+        as well as the new mini-batch of data.
+
+        Like ``fit``, ``partial_fit`` should return the estimator object.
+
+        To clear the model, a new estimator should be constructed, for instance
+        with :func:`base.clone`.
+
+    ``predict``
+        Makes a prediction for each sample. In a :term:`classifier` or
+        :term:`regressor`, this prediction is in the same target space used in
+        fitting (e.g. one of {'red', 'amber', 'green'} if the ``y`` in fitting
+        consisted of these strings).  In a :term:`clusterer` or :term:`outlier
+        detector` the prediction is an integer.
+
+        TODO
+
+        Return type is array ...
+
+        classifier
+            TODO
+
+        clusterer
+            TODO
+            -1 in DBSCAN
+
+        outlier detector
+            TODO
+
+        regressor
+
+        Mention ``return_std``
+
+    ``predict_log_proba``
+        The natural logarithm of the output of :term:`predict_proba`, provided
+        to facilitate numerical stability.
+
+    ``predict_proba``
+        TODO
+
+        Output conventions are like those for ``decision_function`` except
+        in the :term:`binary` case.
+
+        Like other methods, ``predict_proba`` should only be present when the
+        estimator can make probabilistic predictions.  In some cases, this
+        means that the method will only appear after fitting.
+
+    ``score``
+        TODO
+
+    ``score_samples``
+        TODO
+
+    ``set_params``
+        TODO
+
+    ``split``
+        TODO
+
+.. _glossary_parameters:
+
+Parameters
+==========
+
+These common parameter names, specifically used in estimator construction
+(see concept :term:`parameter`) sometimes also appear as function and
+non-estimator parameters with similar semantics.
+
+.. glossary::
+
+    ``class_weight``
+        Used to specify sample weights when fitting classifiers as a function
+        of the :term:`target` class.  Where the :term:`sample_weight`
+        :term:`sample property` is also supported and given, it is multiplied
+        by the ``class_weight`` contribution. Similarly, where ``class_weight``
+        is used in a :term:`multioutput` (including :term:`multilabel`) tasks,
+        the weights are multiplied across outputs (i.e. columns of ``y``).
+
+        By default all samples have equal weight such that classes are
+        effectively weighted by their their prevalence in the training data.
+        This could be achieved explicitly with ``class_weight={label1: 1,
+        label2: 1, ...}`` for all class labels.
+
+        More generally, ``class_weight`` is specified as a dict mapping class
```

Comment:
Do we assume weight is 1 if a class is not present or do we error? (and is there a test ;)